src/assets/src/components/queueSettings.tsx

```diff
@@ -0,0 +1,315 @@
+                                    aria-label='Save Changes'
+                                    onClick={props.onSaveGeneralClick}
+                                >
+                                    Save Changes
+                                </Button>
+                                <Button
+                                    variant='light'
+                                    className={'text-danger ' + buttonSpacing}
+                                    disabled={props.disabled}
+                                    aria-label='Discard Changes'
+                                    onClick={props.onDiscardGeneralClick}
+                                >
+                                    Discard Changes
+                                </Button>
+                            </div>
+                        </Tab.Pane>
+                        <Tab.Pane eventKey='hosts'>
+                            <ManageHostsEditor {...props} />
+                        </Tab.Pane>
+                        <Tab.Pane eventKey='delete'>
+                            <h2>Delete Queue</h2>
+                            <p>Delete the entire queue, including all hosts and current meetings in queue. <strong>This cannot be undone.</strong></p>
+                            <div className='mt-4'>
+                                <Button variant='danger' disabled={props.disabled} aria-label='Delete Queue' onClick={props.onDeleteClick}>
+                                    Delete Queue
+                                </Button>
+                            </div>
+                        </Tab.Pane>
+                    </Tab.Content>
+                </Col>
+            </Row>
+        </Tab.Container>
+    );
+}
+
+interface SettingsPageParams {
+    queue_id: string;
+}
+
+export function ManageQueueSettingsPage(props: PageProps<SettingsPageParams>) {
+    if (!props.user) {
+        redirectToLogin(props.loginUrl);
+    }
+
+    // Set up page state
+    const dialogRef = createRef<Dialog>();
+    const [queue, setQueue] = useState(undefined as QueueHost | undefined);
+    const [queueInitialized, setQueueInitialized] = useState(false);
+    const [authError, setAuthError] = useState(undefined as Error | undefined);
+
+    // Set up WebSocket
+    const queueID = props.match.params.queue_id;
+    if (queueID === undefined) throw new Error("queueID is undefined!");
+    if (!props.user) throw new Error("user is undefined!");
+    const queueIDInt = Number(queueID);
+
+    const setQueueChecked = (q: QueueAttendee | QueueHost | undefined) => {
+        if (!q) {
+            setQueue(q);
+        } else if (isQueueHost(q)) {
+            setQueue(q);
+            setAuthError(undefined);
+        } else {
+            setQueue(undefined);
+            setAuthError(new Error("You are not a host of this queue. If you believe you are seeing this message in error, contact the queue host(s)."));
+        }
+    }
```

Comment:
This is just repeated from `queueManager.tsx` (formerly `edit.tsx`). We discussed earlier that there might be a better way to do this. Let me know if you have any ideas!

Reply:
Off the top of my head, this wrapping could be rolled into a hook (e.g. `useQueueAsHost`) that returns `queue`, `setQueueChecked`, and `authError`. There are probably even better (but more creative and/or change-heavy) solutions. If you want to leave it, I think it's fine for now.

Reply:
I do think this should be cleaned up, but it'd be nice to get this merged as well. The custom hook idea was made a little more complicated as well when I moved the `name`, `description`, and `allowedMeetingTypes` initialization into this block in `queueSettings`. I'm going to create a new issue.

Reply:
See issue #181.